src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sumOfAllNumbers = 0;
+  let copyOfSum;
+
+  const adder = (number = null) => {
+    if (number === null) {
+      copyOfSum = sumOfAllNumbers;
+      sumOfAllNumbers = 0;
+
+      return copyOfSum;
+    } else {
+      sumOfAllNumbers += number;
+    }
```

Comment:
In this part of the code, you don't need to use `else` after `return`. When the function encounters a `return` statement, it stops executing. So if the first `if` condition is met, the function will return and the rest of the code will not be executed.